py_ecc/bn128/bn128_field_elements.py

```diff
@@ -119,14 +119,14 @@ def zero(cls):
 
 
 # Utility methods for polynomial math
-def deg(p):
+def deg(p: Tuple[int, int, int]) -> int:
```

Comment:
It may be prudent for us to define a new `Point` type for this to add an additional level of type safety.

```python
Point = typing.NewType('Point', Tuple[int, int, int])
```

This prevents passing just *any* `(int, int, int)` into these functions, requiring us to be explicit about our point abstraction.

And looking at the code, it appears we'd need both a `Point2D` and a `Point3D` to handle the different dimensionalities that are used.

Reply:
@pipermerriam can you check this changes https://github.com/ethereum/py_ecc/pull/18/commits/e4de2beb46d13c04f0dbd8aae46c681e7a7a2f73